File: test_data/code/python/1380-lucky-numbers-in-a-matrix.py
from typing import List

class Solution:
    def luckyNumbers (self, matrix: List[List[int]]) -> List[int]:
        m, n = len(matrix), len(matrix[0])
        result = []
        
        for i in range(m):
            min_val = min(matrix[i])
            min_idx = matrix[i].index(min_val)
            
            for j in range(n):
                if matrix[i][j] == min_val and max(column(j, matrix)) == min_val:
                    result.append(min_val)
                    break
                    
        return result

def column(j, matrix):
    return [row[j] for row in matrix]

File: test_data/code/java/1380-lucky-numbers-in-a-matrix.java
class Solution {
    public List<Integer> luckyNumbers (int[][] matrix) {
        int m = matrix.length;
        int n = matrix[0].length;
        List<Integer> result = new ArrayList<>();
        for(int i = 0; i < m; i++){
            int min = matrix[i][0];
            int minCol = 0;
            for(int j = 0; j < n; j++){
                if(matrix[i][j] < min){
                    min = matrix[i][j];
                    minCol = j;
                }
            }
            int maxRow = matrix[0][0];
            for(int j = 0; j < n; j++){
                if(matrix[0][j] > maxRow){
                    maxRow = matrix[0][j];
                }
            }
            if(matrix[i][minCol] == maxRow){
                result.add(matrix[i][minCol]);
            }
        }
        return result;
    }
}